### PR TITLE
Feat: [Regions] `contenteditable` content

### PR DESCRIPTION
**Pull request by kakashy**
## Short description
Resolves #3404

## Implementation details
Added a `contenteditable` property to the `content` generated by the `Regions` plugin. This property is `false` by default.

## How to test it
Create a new region and tap/click on the text content of the region.

## Screenshots
image

## Checklist
* [ ] This PR is covered by e2e tests
* [x] It introduces no breaking API changes

### CODE REVIEW

**Comment by katspaugh** [comment]
Looks good to me overall, left a couple of comments.

As an alternative to the `onblur` event you could also use [`onchange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event), I think. Feels a little more appropriate.

File: src/plugins/regions.ts [hunk 1]
```diff
@@ -274,6 +286,16 @@ class SingleRegion extends EventEmitter<RegionEvents> {
     this.emit('update-end')
   }
 
+  private onContentClick(event: MouseEvent) {
+    event.stopPropagation()
+    const contentContainer = event.target as HTMLDivElement
+    contentContainer.focus()
```
Comment (katspaugh) [inline src/plugins/regions.ts:292]: Why is this necessary btw? Isn't focusing the default action on click on editable elements?

File: src/plugins/regions.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+0
```
[the remark this answers is not in the document]
Comment (kakashy) [reply src/plugins/regions.ts:1]: Oh no! That's embarrasing :D

File: src/plugins/regions.ts (hunk 1 above)
Comment (kakashy) [reply src/plugins/regions.ts:295]: True, but with the region container having an event listener for clicks, this (focus on text content) does not happen. The event is immediately propagated to its parent. This function is in hopes that we can step in and handle the content property before propagating the event further. In the latest commit, I have made changes to emit the `region-clicked` event:
```js
private onContentClick(event: MouseEvent) {
    event.stopPropagation()
    const contentContainer = event.target as HTMLDivElement
    contentContainer.focus()
    this.emit('click', event)
  }
  ```

**Comment by kakashy** [comment]
Thanks for the reviews. I have updated my PR with changes from the main repo along with the suggestions in your notes.

One thing that I am still thinking of is the fact that clicking on a text content and clicking away would trigger a `region-updated` event even if the text content remains the same. Perhaps a placeholder variable for the text content could help but this might end up cluttering the plugin file.

**Review by katspaugh — APPROVED** [review]
Thank you!
